x-pack/legacy/plugins/ml/public/data_frame/common/pivot_aggs.ts

```diff
@@ -63,11 +63,39 @@ type PivotAgg = {
 export type PivotAggDict = { [key in AggName]: PivotAgg };
 
 // The internal representation of an aggregation definition.
-export interface PivotAggsConfig {
+export interface PivotAggsConfigBase {
   agg: PIVOT_SUPPORTED_AGGS;
-  field: FieldName;
   aggName: AggName;
   dropDownName: string;
 }
 
+export interface PivotAggsConfigWithUiSupport extends PivotAggsConfigBase {
+  field: FieldName;
+}
+
+export function isPivotAggsConfigWithUiSupport(arg: any): arg is PivotAggsConfigWithUiSupport {
+  return (
+    arg.hasOwnProperty('agg') &&
+    arg.hasOwnProperty('aggName') &&
+    arg.hasOwnProperty('dropDownName') &&
+    arg.hasOwnProperty('field') &&
+    pivotSupportedAggs.includes(arg.agg)
+  );
+}
+
+export type PivotAggsConfig = PivotAggsConfigBase | PivotAggsConfigWithUiSupport;
+
+export type PivotAggsConfigWithUiSupportDict = Dictionary<PivotAggsConfigWithUiSupport>;
 export type PivotAggsConfigDict = Dictionary<PivotAggsConfig>;
+
+export function getEsAggFromAggConfig(groupByConfig: PivotAggsConfigBase): PivotAgg {
+  const esAgg = { ...groupByConfig };
+
+  delete esAgg.agg;
+  delete esAgg.aggName;
+  delete esAgg.dropDownName;
```

Comment:
it looks like this is removing the only three properties in `PivotAggsConfigBase`?
If a `PivotAggsConfigWithUiSupport ` object is passed in, i'll just leave `name` is this correct?
It might be clearer if it created a new object explicitly adding `name`

Reply:
Thanks for pointing that out! It's done this way because a config coming from the advanced editor could have any additional attributes we don't know about, e.g. a `script` attribute in a `terms` agg. That's not properly reflected in `PivotAggsConfigBase` because it should indicate that there could be additional fields. I plan to create more unit tests in a follow up that covers this.